src/pasta/fields/fp.rs

```diff
@@ -16,7 +16,7 @@ use crate::arithmetic::{adc, mac, sbb, FieldExt, Group, SqrtTables};
 // The internal representation of this type is four 64-bit unsigned
 // integers in little-endian order. `Fp` values are always in
 // Montgomery form; i.e., Fp(a) = aR mod p, with R = 2^256.
-#[derive(Clone, Copy, Eq)]
+#[derive(Clone, Copy, Eq, Hash)]
```

Comment:
Why is this `Hash` derive introduced? It trips up clippy and I don't think it's used in this PR.

Reply:
this seems to be used in `pasta::curves`, `impl Hash for $name_affine`, but that in turn doesn't seem to be used anywhere.

Reply:
It is used:
```
   --> src/pasta/curves.rs:656:24
    |
656 |                   self.y.hash(state);
```

Reply:
It was used in a previous iteration. I've removed it.